frontend/src/main.rs

```diff
@@ -321,14 +321,29 @@ fn get_websocket() -> anyhow::Result<WebSocket> {
     let window = web_sys::window().ok_or_else(|| {
         Error::Message("Should be in a web_sys environment to open this web socket".into())
     })?;
-    let host = window
-        .location()
+    let location = window.location();
+    let protocol = match location
+        .protocol()
+        .map_err(|err| Error::Message(format!("Failed to get protocol for window: {err:?}")))?
+        .as_str()
+    {
+        "https:" => Ok("wss"),
+        "http:" => Ok("ws"),
```

Comment:
@savf I was testing for `"http"` and `"https"` instead of what is here now. It seems to work now.